generator/internal/mixin_utils.cc

```diff
@@ -49,10 +49,8 @@ std::unordered_map<std::string, std::string> const& GetMixinProtoPathMap() {
   return *kMixinProtoPathMap;
 }
 
-void ParseHttpRule(std::string const& selector,
-                   YAML::detail::iterator_value const& rule,
-                   google::api::HttpRule& http_rule) {
-  http_rule.set_selector(selector);
+google::api::HttpRule ParseHttpRule(YAML::detail::iterator_value const& rule) {
+  google::api::HttpRule http_rule;
   if (rule["body"]) {
     http_rule.set_body(rule["body"].as<std::string>());
```

Comment:
something about this is goofy. I could just be unfamiliar with the yaml library though.

It seems like either we want `auto& body = rule["body"]` to avoid a double look up.... or the `operator[]` will create the thing if it does not exist. Which seems weird, and seems like we could get rid of the `if` statement.

Reply:
I can avoid double lookup. But if we remove `if`, it throw `unknown file: Failure
C++ exception with description "invalid node; first invalid key: "body"" thrown in the test body.`